osu.Game/Screens/Ranking/Expanded/Statistics/PerformanceStatistic.cs

```diff
@@ -22,30 +26,37 @@ public class PerformanceStatistic : StatisticDisplay
 
         private RollingCounter<int> counter;
 
+        [Resolved]
+        private ScorePerformanceCache performanceCache { get; set; }
+
+        [Resolved]
+        private BeatmapDifficultyCache difficultyCache { get; set; }
+
+        [Resolved]
+        private BeatmapManager beatmapManager { get; set; }
+
         public PerformanceStatistic(ScoreInfo score)
             : base("PP")
         {
             this.score = score;
         }
 
         [BackgroundDependencyLoader]
-        private void load(ScorePerformanceCache performanceCache)
+        private void load()
         {
-            if (score.PP.HasValue)
-            {
-                setPerformanceValue(score.PP.Value);
-            }
-            else
-            {
-                performanceCache.CalculatePerformanceAsync(score, cancellationTokenSource.Token)
-                                .ContinueWith(t => Schedule(() => setPerformanceValue(t.GetResultSafely())), cancellationTokenSource.Token);
-            }
+            new PerformanceBreakdownCalculator(beatmapManager, difficultyCache, performanceCache)
+                .CalculateAsync(score, cancellationTokenSource.Token)
+                .ContinueWith(t => setPerformanceValue(t.GetResultSafely()));
```

Comment:
The schedule on `setPerformanceValue()` that was there on `master` was removed, causing the following error to potentially show up:

```
[runtime] 2022-01-20 21:03:47 [error]: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Cannot mutate the Transforms of a Loaded Drawable while not on the update thread. Consider using Schedule to schedule the mutation operation.)
[runtime] 2022-01-20 21:03:47 [error]: ---> osu.Framework.Graphics.Drawable+InvalidThreadForMutationException: Cannot mutate the Transforms of a Loaded Drawable while not on the update thread. Consider using Schedule to schedule the mutation operation.
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Graphics.Drawable.EnsureMutationAllowed(String member)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Graphics.Transforms.Transformable.AddTransform(Transform transform, Nullable`1 customTransformID)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Graphics.TransformableExtensions.TransformTo[TThis](TThis t, Transform transform)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Graphics.TransformableExtensions.TransformTo[TThis,TValue](TThis t, String propertyOrFieldName, TValue newValue, Double duration, Easing easing)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Game.Graphics.UserInterface.RollingCounter`1.TransformCount(T currentValue, T newValue) in /home/dachb/Dokumenty/opensource/osu/osu.Game/Graphics/UserInterface/RollingCounter.cs:line 162
[runtime] 2022-01-20 21:03:47 [error]: at osu.Game.Graphics.UserInterface.RollingCounter`1.<LoadComplete>b__22_0(ValueChangedEvent`1 val) in /home/dachb/Dokumenty/opensource/osu/osu.Game/Graphics/UserInterface/RollingCounter.cs:line 93
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Bindables.Bindable`1.TriggerValueChange(T previousValue, Bindable`1 source, Boolean propagateToBindings, Boolean bypassChecks)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Bindables.Bindable`1.TriggerValueChange(T previousValue, Bindable`1 source, Boolean propagateToBindings, Boolean bypassChecks)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Framework.Bindables.Bindable`1.set_Value(T value)
[runtime] 2022-01-20 21:03:47 [error]: at osu.Game.Screens.Ranking.Expanded.Statistics.PerformanceStatistic.setPerformanceValue(PerformanceBreakdown breakdown) in /home/dachb/Dokumenty/opensource/osu/osu.Game/Screens/Ranking/Expanded/Statistics/PerformanceStatistic.cs:line 58
[runtime] 2022-01-20 21:03:47 [error]: at osu.Game.Screens.Ranking.Expanded.Statistics.PerformanceStatistic.<load>b__17_0(Task`1 t) in /home/dachb/Dokumenty/opensource/osu/osu.Game/Screens/Ranking/Expanded/Statistics/PerformanceStatistic.cs:line 49
```

Especially visible on maps such as https://osu.ppy.sh/beatmapsets/29157#osu/156352.

Reply:
A careless mistake. Fixed now